src/routes/products/index.svelte

```diff
@@ -68,46 +68,45 @@
 
   let filterElement;
 
-  if (account != null) {
-    balance = account.balance;
-  }
-
-  function updateProducts(filtering) {
-    let queryArgs = new Map();
+  async function updateProducts(filtering) {
+    const query = new Map();
 
     if (filtering.searchQuery) {
-      queryArgs.set('q', filtering.searchQuery);
+      query.set('q', filtering.searchQuery);
     }
     if (filtering.minPrice != null) {
-      queryArgs.set('min_price', filtering.minPrice);
+      query.set('min_price', filtering.minPrice);
     }
     if (filtering.maxPrice != null) {
-      queryArgs.set('max_price', filtering.maxPrice);
+      query.set('max_price', filtering.maxPrice);
     }
-    let excludedColors = filtering.excludedColors.map(color => color.slice(1));
-    if (filtering.colorlessExcluded) {
+
+    const excludedColors = filtering.excludedColors.map(color => color.slice(1));
+    if (!filtering.colorlessIncluded) {
       excludedColors.push(null);
     }
     if (excludedColors.length !== 0) {
-      queryArgs.set('excluded_colors', JSON.stringify(excludedColors));
+      query.set('excluded_colors', JSON.stringify(excludedColors));
+    }
+    query.set('order_by', filtering.order.value.orderBy);
+    query.set('order', filtering.order.value.order);
+    query.set('limit', productLimit);
+    query.set('page', currentPage);
+
+    try {
+      ({ pages, data: products } = await api.json(api.get('/products', { query })));
```

Comment:
Why is the statement wrapped in parentheses?

Reply:
For destructuring. Braces at the beginning of the line are treated as a start of a code block, not an object.